runtime/module.go

```diff
@@ -96,7 +97,22 @@ func ProvideApp() (
 		_, _ = fmt.Fprintln(os.Stderr, err.Error())
 	}
 
-	interfaceRegistry := codectypes.NewInterfaceRegistryWithProtoFiles(protoFiles)
+	interfaceRegistry, err := codectypes.NewInterfaceRegistryWithOptions(codectypes.Options{
+		ProtoFiles:            protoFiles,
+		AddressCodec:          globalAccAddressCodec{},
+		ValidatorAddressCodec: globalValAddressCodec{},
+	})
+	if err != nil {
+		return nil, nil, nil, nil, nil, nil, nil, nil, nil, err
+	}
+
+	// validate the signing context to make sure that messages are properly configured
+	// with cosmos.msg.v1.signer
+	err = interfaceRegistry.SigningContext().Validate()
```

Comment:
Oh... interesting... yeah it shouldn't try to actually decode the address there that's for sure

Reply:
Just pushed something which hopefully will fix this

Reply:
This fix is not necessarily correct though. We should reject empty addresses in any real situation. Not sure what the right way to fix this is. Maybe the validation code is what needs to be fixed rather than something higher up in the API.